src/components/radio/Radio.js

```diff
@@ -161,24 +161,52 @@ export default class RadioComponent extends Field {
   }
 
   setSelectedClasses() {
+    switch (this.inputInfo.attr.type) {
+      case 'checkbox' :
+        this.component.values.forEach(item => {
+          item.selected = this.dataValue[item.value];
+        });
+        break;
+      case 'radio' :
+        this.component.values.forEach(item => {
+          item.selected = item.value === this.dataValue;
+        });
+        break;
+    }
+
     if (this.refs.wrapper) {
       //add/remove selected option class
       const value = this.dataValue;
       const optionSelectedClass = 'radio-selected';
 
       this.refs.wrapper.forEach((wrapper, index) => {
         const input = this.refs.input[index];
-        if (input && input.value.toString() === value.toString()) {
-          //add class to container when selected
-          this.addClass(wrapper, optionSelectedClass);
-          //change "checked" attribute
-          input.setAttribute('checked', 'true');
-        }
-        else {
-          this.removeClass(wrapper, optionSelectedClass);
-          input.removeAttribute('checked');
+        switch (input.type) {
```

Comment:
This seems like the same code but just checks for different "values". Why not just change the code to say this?

```js
const checked = (input.type === 'checkbox') ? value[input.value] : (input.value.toString() === value.toString());
if (checked) {
  this.addClass(wrapper, optionSelectedClass);
  input.setAttribute('checked', 'true');
}
else {
  this.removeClass(wrapper, optionSelectedClass);
  input.removeAttribute('checked');
}
```